src/value.cpp

```diff
@@ -139,6 +139,71 @@ CassError cass_value_get_bytes(const CassValue* value,
   return CASS_OK;
 }
 
+CassError cass_value_get_duration(const CassValue* value, cass_int64_t* months, cass_int64_t* days, cass_int64_t* nanos)
+{
+  cass_int64_t *outs[3];
+  int ctr;
+  size_t data_size = 0;
+  const cass_byte_t* cur_byte = NULL;
+  const cass_byte_t* end = NULL;
+
+  if (value == NULL || value->is_null()) return CASS_ERROR_LIB_NULL_VALUE;
+  if (!cass_value_is_duration(value)) return CASS_ERROR_LIB_INVALID_VALUE_TYPE;
+
+  // Package up the out-args in an array. Duration's always have months, then days, then nanos.
+  outs[0] = months;
+  outs[1] = days;
+  outs[2] = nanos;
+
+  cass_value_get_bytes(value, &cur_byte, &data_size);
+  end = cur_byte + data_size;
+
+  for (ctr = 0; ctr < 3 && cur_byte != end; ++ctr) {
+    int num_extra_bytes;
+    int i;
+    cass_byte_t first_byte = *cur_byte++;
+    if (first_byte <= 127) {
+      // If this is a multibyte vint, at least the MSB of the first byte
+      // will be set. Since that's not the case, this is a one-byte value.
+      *outs[ctr] = first_byte;
+    } else {
+      // The number of consecutive most significant bits of the first-byte tell us how
```

Comment:
Should this logic also be encapsulated in a function `char * decode_vint(char* input, int64_t* out)`?